validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/KeystoreLockerTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.deleteIfStaleLockfileExists;
+import static tech.pegasys.teku.validator.client.loader.KeystoreLocker.longPidToNativeByteArray;
+
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+
+public class KeystoreLockerTest {
+
+  private KeystoreLocker keystoreLocker = new KeystoreLocker();
+
+  @Test
+  void shouldLockKeystoreFileAndFailWhenTryingCreateLockForLockedFile() throws Exception {
+    final Path keystoreFile = Path.of(Resources.getResource("scryptTestVector.json").toURI());
+    deletePastLockfile(keystoreFile);
+
+    Assertions.assertThatCode(() -> keystoreLocker.lockKeystoreFile(keystoreFile))
+        .doesNotThrowAnyException();
+    Assertions.assertThatThrownBy(() -> keystoreLocker.lockKeystoreFile(keystoreFile))
+        .isExactlyInstanceOf(InvalidConfigurationException.class);
+  }
+
+  @Test
+  void doNotDeleteLockfileIfTheProcessIsAlive() throws Exception {
+    final Path keystoreFile = Path.of(Resources.getResource("lockfileTest1.json").toURI());
+    deletePastLockfile(keystoreFile);
+
+    Process process = new ProcessBuilder("/bin/sleep", "5").start();
+    long pid = process.pid();
+    assertThat(process.isAlive()).isTrue();
+    createLockfileWithContent(keystoreFile, longPidToNativeByteArray(pid));
+    assertThat(deleteIfStaleLockfileExists(keystoreFile)).isFalse();
```

Comment:
The delete method should be private and we should just be calling the public `lockKeystore` method and asserting that it either works or throws an exception.  Then check the file exists with the expected content if it should have created the lock (or that the old content exists if it should have failed).

Currently these tests are only checking that the otherwise unused return value is correct - they aren't actually checking the keystore locking really works.

Reply:
Makes sense. Done.